service/invoker/invoker.go

```diff
@@ -188,3 +193,23 @@ func (i *Invoker) Script(height uint64, script []byte, arguments []cadence.Value
 
 	return proc.Value, nil
 }
+
+// ByHeightFrom implements the fvm/env/blocks interface
+func (i *Invoker) ByHeightFrom(height uint64, header *flow.Header) (*flow.Header, error) {
+	if header.Height == height {
+		return header, nil
+	}
+
+	if height > header.Height {
+		minHeight, err := i.index.First()
+		if err != nil {
+			return nil, fmt.Errorf("could not find first indexed height: %w", err)
+		}
+		// imitate flow-go error format
+		msg := fmt.Sprintf("requested height (%d) is not in the range(%d, %d)", height, minHeight, header.Height)
```

Comment:
don't need this extra format step.  just pass the template string and arguments directly into NewValueErrorf

Reply:
Done!